src/libs/dailyBulletin.ts

```diff
@@ -61,50 +64,39 @@ export async function queryLatest() {
 		throw new InternalError('There was a problem getting the most recent email!', e);
 	}
 
-	// Search through the email for any PDF
+	// Get email subject for the date; generate filename from subject
+	const subject = recentMessage.payload?.headers?.filter(h => h.name === 'Subject')[0].value;
+	const filename = generateFilename(
+		subject ?? '',
+		new Date(parseInt(recentMessage.internalDate!, 10))
+	);
+
+	if (!filename) {
+		throw new Error('There was a problem finding the date for the daily bulletin!');
+	}
+
+	// Search through the email for email text
 	const parts = recentMessage.payload!.parts!;
-	let attachmentId: string | null = null;
-	let originalFilename: path.ParsedPath | null = null;
+	let emailText: string | null = null;
 	for (const part of parts) {
-		// If part contains PDF attachment, we're done boys.
-		if (part.mimeType === 'application/pdf' || part.mimeType === 'application/octet-stream') {
-			attachmentId = part.body!.attachmentId!;
-			originalFilename = path.parse(part.filename!);
+		// Take the plaintext of the email and base64 decode
+		if (part.mimeType === 'text/plain' || part.mimeType === 'text/html') {
+			emailText = Buffer.from(part.body?.data ?? '', 'base64').toString('binary');
```

Comment:
ditto here, is this an actual error state you need to handle or were you just trying to get the types to work?

Reply:
In this line I was just trying to get the types to work, I'll change the if statement below to make sure that this error state is handled properly.